src/nickelclock.cc

```diff
@@ -12,89 +12,16 @@
 #include <QMargins>
 #include <QScreen>
 
-#include <NickelHook.h>
-
+#include "nc_common.h"
 #include "nickelclock.h"
 
-const char nc_qt_property[] = "NickelClock";
-const char nc_widget_name[] = "ncTimeLabel";
-
-
-
-NCSettings::NCSettings(QRect const& screenGeom)
-    : settings(NICKEL_CLOCK_DIR "/settings.ini", QSettings::IniFormat)
-{
-    setMaxHMargin(screenGeom);
-    syncSettings();
-}
-
-void NCSettings::syncSettings()
-{
-    settings.sync();
-    QString place = settings.value(placeKey, placeHeader).toString();
-    QString pos = settings.value(posKey, posRight).toString();
-    QString marginStr = settings.value(marginKey, marginAuto).toString();
-    if (place != placeHeader && place != placeFooter)
-        place = placeHeader;
-    if (pos != posLeft && pos != posRight) {
-        pos = posRight;
-    }
-    if (marginStr != marginAuto) {
-        bool ok = false;
-        int margin = marginStr.toInt(&ok);
-        if (!ok || margin > maxHMargin || margin < 0 ) {
-            marginStr = marginAuto;
-        } else {
-            marginStr = QString::number(margin);
-        }
-    }
-    settings.setValue(placeKey, place);
-    settings.setValue(posKey, pos);
-    settings.setValue(marginKey, marginStr);
-    settings.sync();
-}
-
-TimePlacement NCSettings::placement()
-{
-    syncSettings();
-    QString place = settings.value(placeKey).toString();
-    if (place == placeFooter) {
-        return TimePlacement::Footer;
-    } else {
-        return TimePlacement::Header;
-    }
-}
-
-TimePos NCSettings::position()
-{
-    syncSettings();
-    QString pos = settings.value(posKey).toString();
-    if (pos == posLeft) {
-        return TimePos::Left;
-    } else {
-        return TimePos::Right;
-    }
-}
+#include <NickelHook.h>
 
-int NCSettings::hMargin()
-{
-    syncSettings();
-    bool ok = false;
-    int margin = -1;
-    QString marginStr = settings.value(marginKey).toString();
-    if (marginStr != marginAuto) {
-        margin = marginStr.toInt(&ok);
-    }
-    return ok ? margin : -1;
-}
+const char nc_qt_property[] = "NickelClock";
+const char nc_widget_name[] = "ncLabelWidget";
 
-void NCSettings::setMaxHMargin(QRect const& screenGeom)
-{
-    int w = screenGeom.width() < screenGeom.height() ? screenGeom.width()
-                                                     : screenGeom.height();
-    maxHMargin = w / 4;
-    nh_log("screen width: %d, setting margin: %d", w, maxHMargin);
-}
+const char nc_sysfs_gen_bat_cap[] = "/sys/class/power_supply/battery/capacity";
+const char nc_sysfs_mc13892_bat_cap[] = "/sys/class/power_supply/mc13892_bat/capacity";
```

Comment:
@NiLuJe are these the only two paths I have to worry about?

Reply:
So far, yeah ;).